Framework/src/TaskRunner.cxx

```diff
@@ -75,7 +78,7 @@ void TaskRunner::init(InitContext& iCtx)
   ILOG(Info) << "initializing TaskRunner" << ENDM;
 
   // registering state machine callbacks
-  iCtx.services().get<CallbackService>().set(CallbackService::Id::Start, [this]() { start(); });
+  iCtx.services().get<CallbackService>().set(CallbackService::Id::Start, [this, &options = iCtx.options()]() { start(options); });
```

Comment:
This assumes that the reference to `iCtx.options()` will be still valid during the start. This isn't completely safe, but I guess we don't have other choice for now.

Reply:
Indeed, we don't.